packages/shared/src/api/channelContentConfig.ts

```diff
@@ -0,0 +1,79 @@
+export enum CollectionRendererId {
+  notebook = 'tlon.r0.collection.notebook',
+  chat = 'tlon.r0.collection.chat',
+  gallery = 'tlon.r0.collection.gallery',
+}
+
+export enum DraftInputId {
+  notebook = 'tlon.r0.input.notebook',
+  chat = 'tlon.r0.input.chat',
+  gallery = 'tlon.r0.input.gallery',
+}
+
+export enum PostContentRendererId {
+  notebook = 'tlon.r0.content.notebook',
+  chat = 'tlon.r0.content.chat',
+  gallery = 'tlon.r0.content.gallery',
+}
+
+/**
+ * Configures the custom components used to create content in a channel.
+ */
+export interface ChannelContentConfiguration {
+  /**
+   * Which controls are available when composing a new post?
+   */
+  draftInput: DraftInputId;
```

Comment:
I'd like to make `draftInput` and `defaultPostContentRenderer` plural, but we're missing some stuff to make that happen:
- We don't have UI to change the draft input if there is more than one available (we could expand the gallery action FAB -> action sheet paradigm for this)
- I'd like to be able to say "this kind of post should be rendered with content renderer A, and this _other_ kind of post should be rendered with renderer B" – but it's not clear how to distinguish "kinds" of posts.

Reply:
/1/chan/chat/~bitpyx-dildus/interface/msg/170141184507040415784022802543350382592

![Screenshot 2024-10-21 at 9 02 39 AM](https://github.com/user-attachments/assets/f91e4c33-2688-41e1-b9ef-3612d856efab)
![Screenshot 2024-10-21 at 9 02 52 AM](https://github.com/user-attachments/assets/47c5f978-77a2-43af-baa5-56ad1b36e655)

Reply:
I'm not sure what form of this API we want to ship as a prototype (e.g. should we ship the structured description payload, should we namespace stuff as prototype, etc); but if we want some kind of stability here, I'd say we should change these fields to allow for plurality, but only handle values as singles:

```ts
defaultPostContentRenderers: { '*': 'tlon.r0.content.chat' },
draftInputs: ['tlon.r0.input.chat'], // or the dictionary approach i mentioned in chat above
```